lib/integrations/Exchange/components/AddExchangeIntegration.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState } from "react";
+import { useForm } from "react-hook-form";
+
+import {
+  DialogHeader,
+  DialogProps,
+  Dialog,
+  DialogContent,
+  DialogClose,
+  DialogFooter,
+} from "@components/Dialog";
+import { Form, TextField } from "@components/form/fields";
+import { Alert } from "@components/ui/Alert";
+import Button from "@components/ui/Button";
+
+type Props = {
+  onSubmit: () => void;
+};
+
+export const ADD_EXCHNAGE_INTEGRATION_FORM_TITLE = "addExchange";
+export type AddExchangeIntegrationRequest = {
+  url: string;
+  username: string;
+  password: string;
+};
+
+export function AddExchangeIntegrationModal(props: DialogProps) {
+  const form = useForm({
+    defaultValues: {
+      url: process.env.EXCHANGE_DEFAULT_EWS_URL || "",
+      username: "",
+      password: "",
+    },
+  });
+  const [errorMessage, setErrorMessage] = useState("");
+  return (
+    <Dialog {...props}>
+      <DialogContent>
+        <DialogHeader
+          title="Connect to Exchange Server"
```

Comment:
These should ideally be extracted as translation keys. I'm happy to do this if you'd like

Reply:
yeah sure 👍